src/Services/Masa.Auth.Service.Admin/Application/Organizations/CommandHandler.cs

```diff
@@ -10,29 +10,47 @@ public CommandHandler(IDepartmentRepository departmentRepository)
     }
 
     [EventHandler]
-    public async Task CreateDepartmentAsync(AddDepartmentCommand createDepartmentCommand)
+    public async Task AddDepartmentAsync(AddDepartmentCommand createDepartmentCommand)
     {
-        var parent = await _departmentRepository.FindAsync(createDepartmentCommand.ParentId);
-        var department = new Department(createDepartmentCommand.Name, createDepartmentCommand.Description,
-            parent, createDepartmentCommand.Enabled, createDepartmentCommand.StaffIds.ToArray());
-
-        await _departmentRepository.AddAsync(department);
+        var dto = createDepartmentCommand.AddOrUpdateDepartmentDto;
+        var parent = await _departmentRepository.FindAsync(dto.ParentId);
+        // Add
+        if (dto.Id == Guid.Empty)
```

Comment:
1. Abstract a `BaseUpsertDto` or `BaseAddOrUpdateDto`
2. Add `InferredAction` help determine whether to add or update, maybe you need JsonIgnore this property
3. Finally, use switch or if/else process logic